alacritty/src/config/bindings.rs

```diff
@@ -61,16 +61,21 @@ impl<T: Eq> Binding<T> {
             return false;
         }
 
-        // Completely empty modes match all modes.
-        if (self.mode.is_empty() && self.notmode.is_empty())
-            || (binding.mode.is_empty() && binding.notmode.is_empty())
-        {
-            return true;
+        let selfmode = if self.mode.is_empty() { TermMode::ANY } else { self.mode };
+
+        let bindingmode = if binding.mode.is_empty() { TermMode::ANY } else { binding.mode };
+
+        if !selfmode.intersects(bindingmode) {
+            return false;
+        }
+
+        // If the binding requires a mode while the other binding is never bound, they are not in
+        // conflict.
```

Comment:
```suggestion
        // The bindings are never active at the same time when the required modes of one binding are part of the forbidden bindings of the other.
```

I think that's a little easier to understand.